src/audio/sink_api_helper.c

```diff
@@ -175,3 +175,18 @@ int sink_set_alignment_constants(struct sof_sink __sparse_cache *sink,
 		return sink->ops->set_alignment_constants(sink, byte_align, frame_align_req);
 	return 0;
 }
+
+void sink_set_min_free(struct sof_sink __sparse_cache *sink, size_t min_free)
+{
+	sink->min_free = min_free;
+}
+
+size_t sink_get_min_free(struct sof_sink __sparse_cache *sink)
+{
+	return sink->min_free;
+}
+
+bool sink_get_is_enough_free(struct sof_sink __sparse_cache *sink)
+{
+	return (sink->min_free <= sink_get_free_size(sink));
+}
```

Comment:
These are better as static inline in the header as it will save a call() in most cases.

Reply:
see comments for btian:

> No, I can't do it like this
Sink/source is a structure that must not be accessible directly (like private in C++). Therefore in sink_api.h the structure is not defined here, all the user has is a forward declaration.
Operations on the structure must be encapsulated in separate C file.

Reply:
ok, so these are private for core use only. Zephyr takes an approach based on API prefix, but we could do that OR we can
1) Use existing ```include/sof``` as private API directory
2) Use new ```include/public``` as public API directory

We could start moving APIs to public quite quickly and this would force modules on the public API ?

Reply:
as long as these functions are not called in timing critical path, it is fine for the price, if in copy/process function, then we may need move it further.

Reply:
I'm not in favour of having extra calls as the cost of partitioning public/private APIs. We should be able to provide an optimal and clear client API using doxygen, comment partionint and get/set accessors.

Reply:
@lgirdwood @marcinszkudlinski If these are truly general (usable for any sink/source implementation), then we can indeed inline them.

For source/sink methods that may be overridden by the specific type of source/sink, we have to incur the function call cost, as this is only way to abstract the interface.

But yeah. we can mix the two. True generic helpers can be inlined, while the "ops" methods incur a call overhead.

Reply:
@kv2019i @lgirdwood  we'll need to expose sink / source structure
there was a lot effort NOT to do this
all sink/src functions are already implemented in this way, please don't change it for those 2

Reply:
So everything is exposed anyway (open code), but stating manadory usage of accessors in the header would be helpful to eliminate out of scope usage.